### PR TITLE
refactor(frontend): use iceberg-rust instead of icelake to enumerate iceberg files and snapshots.

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- https://github.com/risingwavelabs/risingwave/issues/17548
- Use iceberg-rust to enumerate iceberg files and snapshots.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/source/iceberg/mod.rs [hunk 1]
```diff
@@ -171,58 +172,64 @@ impl IcebergSplitEnumerator {
...
-            )
-            .await?
-        {
-            if file.content != DataContentType::Data {
-                bail!("Reading iceberg table with delete files is unsupported. Please try to compact the table first.");
+
+        let snapshot = table
+            .metadata()
+            .snapshot_by_id(snapshot_id)
+            .expect("snapshot must exists");
```
Comment (fuyufjh) [inline src/connector/src/source/iceberg/mod.rs:212]:
```suggestion
            .expect("snapshot must exist");
```

**Review by fuyufjh — APPROVED** [review]
LGTM

**Review by xxchan — APPROVED** [review]

**Comment by chenzl25** [comment]
It seems iceberg-rust introduce some breaking change on `s3a://` and `s3://` protocol
```
DataInvalid => Invalid s3 url: s3a://iceberg/demo_db/e2e_demo_table/metadata/snap-1-1-7b361ebc-1b38-4dee-aa15-eeb7f9416b63.avro, should start with s3://iceberg/
```

**Comment by chenzl25** [comment]
> It seems iceberg-rust introduces some breaking changes on `s3a://` and `s3://` protocols
> 
> ```
> DataInvalid => Invalid s3 url: s3a://iceberg/demo_db/e2e_demo_table/metadata/snap-1-1-7b361ebc-1b38-4dee-aa15-eeb7f9416b63.avro, should start with s3://iceberg/
> ```

https://github.com/apache/iceberg-rust/issues/434

**Comment by chenzl25** [comment]
> > It seems iceberg-rust introduces some breaking changes on `s3a://` and `s3://` protocols
> > ```
> > DataInvalid => Invalid s3 url: s3a://iceberg/demo_db/e2e_demo_table/metadata/snap-1-1-7b361ebc-1b38-4dee-aa15-eeb7f9416b63.avro, should start with s3://iceberg/
> > ```
> 
> [apache/iceberg-rust#434](https://github.com/apache/iceberg-rust/issues/434)

This is an expected behavior. Let's move on.

**Pull request merged**